code/game/objects/items/weapons/twohanded.dm

```diff
@@ -373,7 +373,8 @@
 		if(!isliving(target))
 			return
 		else
-			target.Weaken(4)
+			if(!istype(src, /obj/item/weapon/twohanded/chainsaw/rubber))

```

Comment:
It should really only be a matter of defining an `obj/item/weapon/twohanded/toy/chainsaw`, copying over the chainsaw's `update_icon()` proc, and copying over the sound handling into the toy chainsaw's `attack` proc. Still has a bit of copypasta, but that can't really be avoided, and is better than snowflaking it into here.